runtime/signals.c

```diff
@@ -234,7 +234,7 @@ void caml_request_minor_gc (void)
 
 CAMLexport int caml_check_pending_actions(void)
 {
-  return (caml_check_pending_interrupt() ||
+  return (Caml_check_gc_interrupt(Caml_state) ||
           caml_check_pending_signals());
```

Comment:
If `Caml_check_gc_interrupt` is sufficient in POPTRAP and CHECK_SIGNALS in the bytecode then we probably don't need to `caml_check_pending_signals` here either. Doesn't hurt though.

Reply:
Currently there are races (e.g. allocation starts -> signal arrives -> allocation ends) so I prefer if `caml_check_pending_actions` is more correct until this is fixed.
I wonder if these races threaten the interpreter (i.e. risk of completely missing a signal)? One needs to reintroduce `caml_update_young_limits` to avoid this.

Reply:
Signals are checked with process_pending_actions during allocations in the interpreter so it seems that the risk of completely missing a signal only concerns e.g. loops that do not allocate.

Reply:
The fix for this (in general) is to test for pending signals and leave young_limit at INTERRUPT_MAGIC if that's the case?

Reply:
Yes that's essentially what `caml_update_young_limit` did. But I am not sure it is straightforward to reintroduce it (cf. loop in Alloc_small.)
Now looking at Alloc_small again and considering that it is called with GC=process_pending_actions, I do not think that the race I mentioned applies to the interpreter. I think it's ok as it is.

Reply:
To sum up, it is not ok to remove check_pending_signals here, but it is ok to remove it in the interpreter because it calls process_pending_signals unconditionally after every allocation